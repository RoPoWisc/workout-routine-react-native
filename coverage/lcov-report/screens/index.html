
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.15% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>107/296</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.28% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/61</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">24.32% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>27/111</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.77% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>107/291</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="AddWorkout.js"><a href="AddWorkout.js.html">AddWorkout.js</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="EditWorkout.js"><a href="EditWorkout.js.html">EditWorkout.js</a></td>
	<td data-value="55.56" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 55%"></div><div class="cover-empty" style="width: 45%"></div></div>
	</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file low" data-value="ForgotPassword.js"><a href="ForgotPassword.js.html">ForgotPassword.js</a></td>
	<td data-value="42.11" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 42%"></div><div class="cover-empty" style="width: 58%"></div></div>
	</td>
	<td data-value="42.11" class="pct low">42.11%</td>
	<td data-value="19" class="abs low">8/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="42.11" class="pct low">42.11%</td>
	<td data-value="19" class="abs low">8/19</td>
	</tr>

<tr>
	<td class="file medium" data-value="Home.js"><a href="Home.js.html">Home.js</a></td>
	<td data-value="55.56" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 55%"></div><div class="cover-empty" style="width: 45%"></div></div>
	</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="27" class="abs medium">15/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="27" class="abs medium">15/27</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="40" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 40%"></div><div class="cover-empty" style="width: 60%"></div></div>
	</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="30" class="abs low">12/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="30" class="abs low">12/30</td>
	</tr>

<tr>
	<td class="file low" data-value="PasswordReset.js"><a href="PasswordReset.js.html">PasswordReset.js</a></td>
	<td data-value="26.09" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 26%"></div><div class="cover-empty" style="width: 74%"></div></div>
	</td>
	<td data-value="26.09" class="pct low">26.09%</td>
	<td data-value="23" class="abs low">6/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="26.09" class="pct low">26.09%</td>
	<td data-value="23" class="abs low">6/23</td>
	</tr>

<tr>
	<td class="file medium" data-value="Progress.js"><a href="Progress.js.html">Progress.js</a></td>
	<td data-value="62.5" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 62%"></div><div class="cover-empty" style="width: 38%"></div></div>
	</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="16" class="abs medium">10/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="16" class="abs medium">10/16</td>
	</tr>

<tr>
	<td class="file low" data-value="Register.js"><a href="Register.js.html">Register.js</a></td>
	<td data-value="29.63" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 29%"></div><div class="cover-empty" style="width: 71%"></div></div>
	</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	</tr>

<tr>
	<td class="file low" data-value="Settings.js"><a href="Settings.js.html">Settings.js</a></td>
	<td data-value="46.15" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 46%"></div><div class="cover-empty" style="width: 54%"></div></div>
	</td>
	<td data-value="46.15" class="pct low">46.15%</td>
	<td data-value="13" class="abs low">6/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="46.15" class="pct low">46.15%</td>
	<td data-value="13" class="abs low">6/13</td>
	</tr>

<tr>
	<td class="file low" data-value="Workout.js"><a href="Workout.js.html">Workout.js</a></td>
	<td data-value="16" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 16%"></div><div class="cover-empty" style="width: 84%"></div></div>
	</td>
	<td data-value="16" class="pct low">16%</td>
	<td data-value="100" class="abs low">16/100</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="25" class="abs low">0/25</td>
	<td data-value="3.03" class="pct low">3.03%</td>
	<td data-value="33" class="abs low">1/33</td>
	<td data-value="16.84" class="pct low">16.84%</td>
	<td data-value="95" class="abs low">16/95</td>
	</tr>

<tr>
	<td class="file medium" data-value="initial.js"><a href="initial.js.html">initial.js</a></td>
	<td data-value="65.22" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 65%"></div><div class="cover-empty" style="width: 35%"></div></div>
	</td>
	<td data-value="65.22" class="pct medium">65.22%</td>
	<td data-value="23" class="abs medium">15/23</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="65.22" class="pct medium">65.22%</td>
	<td data-value="23" class="abs medium">15/23</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Nov 13 2020 18:31:04 GMT-0600 (Central Standard Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    