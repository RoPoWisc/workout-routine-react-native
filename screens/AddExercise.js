import React, { useState } from 'react';
import { View, ImageBackground, StyleSheet, TouchableOpacity, Image, TextInput } from 'react-native'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { } from '../actions/user'
import {
    ApplicationProvider,
    Button,
    Icon,
    IconRegistry,
    Layout,
    Text,
    CheckBox,
    List
} from '@ui-kitten/components';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import * as eva from '@eva-design/eva';
import { set } from 'react-native-reanimated';
import { vw, vh, vmin, vmax } from 'react-native-expo-viewport-units'

const MenuIcon = (props) => (
    <Icon {...props} name='menu-outline' />
);

const CollapseIcon = (props) => (
    <Icon {...props} name='arrow-ios-upward-outline' />
);

const ExpandIcon = (props) => (
    <Icon {...props} name='arrow-ios-downward-outline' />
);

const RemoveIcon = (props) => (
    <Icon {...props}  name='close-circle-outline' />
  );

const exerciseStyles = StyleSheet.create({
    exercise: {
        backgroundColor: '#E5E5E5',
        width: '93%',
        borderRadius: 10,
        padding: 5,
        margin: 5
    },
    header: {
        fontSize: 30
    },
    exerciseInfo: {
        fontSize: 20
    },
    weight: {
        fontSize: 20,
        paddingLeft: 10
    },
    reps: {
        position: 'absolute',
        top: '12%',
        fontSize: 20,
        left: '38%'
    }
});

class AddExercise extends React.Component {

    constructor(props) {
        super(props)
        this.addSetHandler = this.addSetHandler.bind(this);
        this.changeTextHandler = this.changeWeightHandler.bind(this);
        this.changeRepsHandler = this.changeRepsHandler.bind(this);
        this.getDate = this.getDate.bind(this);
        this.removeSetHandler = this.removeSetHandler.bind(this);
        this.changeRoutineNameHandler = this.changeRoutineNameHandler.bind(this);
        this.changeRoutineDayHandler = this.changeRoutineDayHandler.bind(this);

        // this.state = {
        //     workout: {
        //         routineName: this.props.navigation.state.params.workoutData.routineName,
        //         routineDay: this.props.navigation.state.params.workoutData.routineDay,
        //         exerciseArray: this.props.navigation.state.params.workoutData.exerciseArray
        //     }
        // }
    }

    componentDidMount = async () => {

        let url = 'https://workout-routine-builder-api.herokuapp.com/exercises/public/'
        let bearer = 'Bearer ' + this.props.user.bearerToken;

        try {
            let response = await fetch(url, {
                method: 'PATCH',
                headers: {
                    Accept: '/',
                    'Content-Type': 'application/json',
                    'Authorization': bearer
                },
            });
            let responseJson = await response.json();
            console.log(responseJson);
        } catch (e) {
            alert(e);
        }
        
        
        

        //this.props.navigation.navigate('DrawerNavigator')
    }

    getDate() {
        var date = new Date();
        var time = (new Date(date)).getTime();
        return time;
    }

    endWorkoutHandler = async () => {
        let bodyJSON = JSON.stringify({
            'routineName': this.state.workout.routineName,
                    'routineDay': this.state.workout.routineDay,
                    'exerciseArray': this.state.workout.exerciseArray
        });

        let url = 'https://workout-routine-builder-api.herokuapp.com/users/' + this.props.user.userServer['_id'] + '/add/workout'
        let bearer = 'Bearer ' + this.props.user.bearerToken;

        try {
            let response = await fetch(url, {
                method: 'PATCH',
                headers: {
                    Accept: '/',
                    'Content-Type': 'application/json',
                    'Authorization': bearer
                },
                body: bodyJSON
            });
        } catch (e) {
            alert(e);
        }

        this.props.navigation.navigate('DrawerNavigator')
	}

    addSetHandler(exercise_object) {
        exercise_object.sets.push({weight: 0, reps: 0, checked: 'false', id: this.getDate()});

        let newArray = this.state.workout.exerciseArray.map(element => element.exercise_id == exercise_object.exercise_id ? { ...element, sets: exercise_object.sets } : element);

        let state = {
            workout: {
                ...this.state.workout,
                exerciseArray: newArray,
            }
        }
        this.setState({ workout: state.workout }, //console.log(this.state.workout.exerciseArray)
        );
    }

    compareExerciseObjects(obj1, obj2) {
        if (obj1.name != obj2.name || obj1.totalVolume != obj2.totalVolume || obj1.show != obj2.show) {
            return false;
        }

        const keys = Object.keys(obj1);
        for (let key of keys) {
            if (obj1[key] != obj2[key]) {
                return false;
            }
        }

        return true;
    }

    Check = (props) => {
        const [checked, setChecked] = React.useState(props.checked);

        // TODO: Change loops to use hashmap to improve checkbox performance

        const isCheckedHandler = () => {
            let workoutCopy = this.state.workout;
            let i = 0;
            let j = 0;
            for (let exercise of this.state.workout.exerciseArray) {
                if (props.exercise_id == exercise.exercise_id) {
                    for (let entry of exercise.sets) {
                        if (entry.id == props.id) {
                            workoutCopy.exerciseArray[i].sets[j] = { ...workoutCopy.exerciseArray[i].sets[j], checked: !workoutCopy.exerciseArray[i].sets[j].checked }
                            setChecked(workoutCopy.exerciseArray[i].sets[j].checked);
                            break;
                        }
                        ++j;
                    }
                    break;
                }
                ++i;
            }
            this.setState({ workout: workoutCopy }, () => {}//console.log('updated state\n', workoutCopy)
            );
        };

        return (
            <CheckBox
                checked={checked}
                onChange={nextChecked => isCheckedHandler()}
                style={{
                    width: 75,
                }}
                status='success'>
            </CheckBox>
        );
    }

    changeWeightHandler(exercise_entry, text) {

        let newSetsArray;
        let newState = {
            ...this.state.workout
        }

        let i = 0;
        for (var exercise of this.state.workout.exerciseArray) {
            if (exercise.exercise_id == exercise_entry.exercise_id) {
                newSetsArray = exercise.sets.map(set => set.id == exercise_entry.id ? {
                    weight: text, reps: set.reps, checked: set.checked, id: set.id
                } : set);
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i],
                    sets: newSetsArray
                }
            } else {
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i]
                }
            }
            ++i;
        }

        this.setState({ workout: newState });
    }

    changeRepsHandler(exercise_entry, text) {

        let newSetsArray;
        let newState = {
            ...this.state.workout
        }

        let i = 0;
        for (var exercise of this.state.workout.exerciseArray) {
            if (exercise.exercise_id == exercise_entry.exercise_id) {
                newSetsArray = exercise.sets.map(set => set.id == exercise_entry.id ? {
                    weight: set.weight, reps: text, checked: set.checked, id: set.id
                } : set);
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i],
                    sets: newSetsArray
                }
            } else {
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i]
                }
            }
            ++i;
        }
      
        this.setState({workout: newState});
    }

    removeSetHandler(exercise_entry, exercise_id) {
        // console.log(exercise_id);
        // console.log(exercise_entry);
        let newState = {
            ...this.state.workout
        }

        let i = 0;
        for (var exercise of this.state.workout.exerciseArray) {
            if (exercise.exercise_id == exercise_entry.exercise_id) {
                let newSetsArray = [...exercise.sets];
                let j = 0;
                console.log(exercise.sets);
                for (var set of exercise.sets) {
                    if (set.id == exercise_entry.id) {
                        break;
                    }
                    ++j;
                }
                newSetsArray.splice(j, 1);
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i],
                    sets: newSetsArray
                }
            } else {
                newState.exerciseArray[i] = {
                    ...this.state.workout.exerciseArray[i]
                }
            }
            ++i;
        }

        //console.log(newState);

        this.setState({ workout: newState });
    }

    changeRoutineNameHandler(name) {
        let newState = {
            ...this.state.workout
        }
        newState.routineName = name;

        console.log(name);

        this.setState({workout: newState})
    }

    changeRoutineDayHandler(day) {
        let newState = {
            ...this.state.workout
        }
        newState.routineDay = day;

        this.setState({workout: newState})
    }

    render() {

        const ExerciseEntry = (exercise_entry, exercise_id) => {
            const [weightText, setWeight] = React.useState('Weight');
            const [repText, setRep] = React.useState('Reps');

            return (
                <View style={{
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    backgroundColor: '#0466C880',
                    borderRadius: 8,
                    paddingTop: 3,
                    paddingBottom: 3,
                    marginVertical: 3,
                    paddingRight: 10
                }}>
                    <TextInput category='s1' style={exerciseStyles.weight} onChangeText={text => setWeight(text)} onSubmitEditing={() => this.changeWeightHandler(exercise_entry, weightText)} >{exercise_entry.weight}</TextInput>
                    <TextInput category='s1' style={exerciseStyles.reps} onChangeText={text => setRep(text)} onSubmitEditing={() => this.changeRepsHandler(exercise_entry, repText)}>{exercise_entry.reps}</TextInput>
                    <TouchableOpacity onPress={() => this.removeSetHandler(exercise_entry, exercise_entry.exercise_id)}>
                        <Icon style={{width: 25, height: 25}} fill='#DB504A' name='close-circle-outline' />
                    </TouchableOpacity>
                    
                </View>
            );
        };

        const Exercise = (exercise_object) => {

            const [showList, setShowList] = React.useState(true);

            const showElementHandler = () => {
                setShowList(!showList);
            };

            return (
                <View style={exerciseStyles.exercise}>
                    <View style={{
                        justifyContent: 'space-between',
                        flexDirection: 'row',
                        borderBottomColor: '#002855',
                        borderBottomWidth: 1,
                    }}>
                        <Text category='s1' style={exerciseStyles.header}>{exercise_object.name}</Text>
                        <Text category='s1' style={exerciseStyles.header}>{exercise_object.totalVolume}</Text>
                    </View>

                    { showList && <View>
                        <View style={{
                            flexDirection: 'row',
                            justifyContent: 'space-between',
                        }}>
                            <Text category='s1' style={exerciseStyles.exerciseInfo}>Weight</Text>
                            <Text category='s1' style={exerciseStyles.exerciseInfo}>Reps</Text>
                            <Text category='s1' style={exerciseStyles.exerciseInfo}>Remove Set</Text>
                        </View>

                        <List
                            style={{ backgroundColor: '#E5E5E5' }}
                            data={exercise_object.sets}
                            renderItem={({ item }) => (
                                <ExerciseEntry weight={item.weight} reps={item.reps} checked={item.checked} id={item.id} exercise_id={exercise_object.exercise_id}></ExerciseEntry>
                            )}
                        />
                        <Button style={{ marginVertical: 3 }} appearance={'outline'} size={'small'} onPress={() => this.addSetHandler(exercise_object)}>Add Set</Button>
                    </View>}
                    <Button appearance='ghost' accessoryLeft={showList ? CollapseIcon : ExpandIcon} size='small' style={{ margin: -8 }} onPress={() => showElementHandler(exercise_object.exercise_id)}></Button>
                </View>
            );
        };

        return (
            <>
                <IconRegistry icons={EvaIconsPack} />
                <ApplicationProvider {...eva} theme={eva.light}>
                    <Layout style={styles.container}>
                        
                    </Layout>
                </ApplicationProvider>
            </>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 30,
        //justifyContent: 'center',
        alignItems: 'center',
    },
    button2: {
        margin: 10,
        width: '80%',
        //backgroundColor: '#DB3020',
        borderRadius: 8  
    },
    button: {
        margin: 20,
        marginTop: 10,
        width: '80%',
        backgroundColor: '#DB504A',
        borderColor: '#DB504A',
        borderRadius: 8
    },
    header: {
        padding: 10,
        paddingLeft: 15,
        alignSelf: "flex-start",
        width: '100%'
    },
    topBar: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    name: {
        color: 'gray',
        fontSize: 28,
        fontWeight: 'bold'
    },
    settingsButton: {
        //alignSelf: 'flex-end',
        margin: 2
    },
    text: {
        //textAlign: 'center',
        fontSize: 30,
        fontWeight: 'bold'
    },
    likeButton: {
        marginVertical: 16,
    },
    optionButton: {
        marginTop: vh(1.2),
        marginRight: vw(2),
        height: vh(7),
        width: vw(14),
    },
});
const mapDispatchToProps = dispatch => {
    return bindActionCreators({}, dispatch)
}

const mapStateToProps = state => {
    return {
        user: state.user
    }
}

export default connect(
    mapStateToProps
)(AddExercise)