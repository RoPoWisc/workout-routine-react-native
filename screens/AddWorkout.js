import React from 'react';
import { StyleSheet, TouchableOpacity, Image } from 'react-native';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { } from '../actions/user'
import {
  ApplicationProvider,
  Button,
  Icon,
  IconRegistry,
  Layout,
  Text,
} from '@ui-kitten/components';
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import * as eva from '@eva-design/eva';

const AddIcon = (props) => (
  <Icon {...props} name='plus-outline'/>
);

class Home extends React.Component {

    render() {
        return (
            <>
                <IconRegistry icons={EvaIconsPack}/>
                <ApplicationProvider {...eva} theme={eva.light}>
                  <Layout style={styles.header}>
                    <Layout style={styles.headerLeft}>
                      <Text style={styles.textMain} category='s1'>
                        Add
                      </Text>
                      <Text style={styles.textSub} appearance='hint'>
                        Workout
                      </Text>
                    </Layout>
                    <Layout style={styles.headerRight}>
                      <TouchableOpacity>
                        {/* <Icon
                          name='menu-outline'
                        /> */}
                        <Image style={styles.menuImage} source={require('../assets/logo1.png')}/>
                      </TouchableOpacity>
                    </Layout>
                  </Layout>
                  <Layout style={styles.container}>
                    <Layout style={styles.workout}>
                      <Layout style={styles.headerLeft}>
                        <Text style={styles.textMain} category='s1'>
                          Prebuilt
                        </Text>
                        <Text style={styles.textSub} appearance='hint'>
                          Workout
                        </Text>
                      </Layout>
                      <Layout style={styles.headerRight}>
                        <Button style={styles.button} size='giant' accessoryLeft={AddIcon}/>
                    </Layout>
                    </Layout>
                    <Layout style={styles.workout}>
                      <Layout style={styles.headerLeft}>
                        <Text style={styles.textMain} category='s1'>
                          Custom
                        </Text>
                        <Text style={styles.textSub} appearance='hint'>
                          Workout
                        </Text>
                      </Layout>
                      <Layout style={styles.headerRight}>
                        <Button style={styles.button} status='danger' size='giant' accessoryLeft={AddIcon}/>
                    </Layout>
                    </Layout>
                  </Layout>
                </ApplicationProvider>
            </>
        )
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 3,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  header: {
    //borderWidth: 1,
    flex: 1,
    flexDirection: 'row',
    marginTop: 25,
    alignItems: 'center',
  },
  headerLeft: {
    //borderWidth: 1,
    flex: 3,
    alignItems: 'flex-start',
    marginBottom: 40,
    marginLeft: 20,
  },
  headerRight: {
    //borderWidth: 1,
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'flex-start',
    justifyContent: 'flex-end',
    marginRight: 5,
  },
  textMain:{
    fontSize: 40,
    fontWeight: 'bold',
  },
  textSub: {
    fontSize: 30,
  },
  text: {
    fontSize: 20,
    alignSelf: 'flex-start',
    flex: 3,
    marginLeft: 20,
  },
  menuImage: {
    position: 'relative',
    borderRadius: 20,
    width: 80,
    height: 80,
    resizeMode: 'contain',
  },
  likeButton: {
    marginVertical: 16,
  },
  workout: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  button: {
    marginRight: 40,
    marginTop: 35,
    borderRadius: 30
  },
});
const mapDispatchToProps = dispatch => {
	return bindActionCreators({}, dispatch)
}

const mapStateToProps = state => {
	return {
		user: state.user
	}
}

export default connect(
    mapStateToProps
)(Home)