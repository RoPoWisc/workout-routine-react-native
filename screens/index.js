import React from 'react'

export const YourProgress = ({navigation}) => <YourProgress navigation={navigation} name="Your Progress" />
export const AddWorkout = ({navigation}) => <AddWorkout navigation={navigation} name="Add Workout" />
export const EditWorkout = ({navigation}) => <EditWorkout navigation={navigation} name="Edit Workout" />
export const Settings = ({navigation}) => <Settings navigation={navigation} name="Settings" />
//export const Login = ({navigation}) => <Login navigation={navigation} name="Login" />
