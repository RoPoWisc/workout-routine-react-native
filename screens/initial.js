import React from 'react'
import { View, ImageBackground, StyleSheet, TouchableOpacity, Text, Image } from 'react-native'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { updateEmail, updatePassword, fetchUserObj } from '../actions/user'
import { vw, vh} from 'react-native-expo-viewport-units'
import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';


function cacheImages(images) {
	return images.map(image => {
	  if (typeof image === 'string') {
		return Image.prefetch(image);
	  } else {
		return Asset.fromModule(image).downloadAsync();
	  }
	});
  }
  
export class Initial extends React.Component {
	state = {
		isReady: false,
	  };
	  
	  async _loadAssetsAsync() {
		const imageAssets = cacheImages([
		  require('../assets/home.jpg'),
		  require('../assets/home1.jpg'),
		  require('../assets/curls.jpg'),
		  require('../assets/initScreen.jpg'),
		  require('../assets/logo-wh.png'),
		  require('../assets/register.jpg'),
		  require('../assets/logo1.png'),
		  require('../assets/loginbackground.jpeg'),
		  require('../assets/ForgotPwd.jpg')
		]);
		
		await Promise.all([...imageAssets]);
	  }
	
	componentDidMount = async () => {
		try {
			if(this.props.user.userId !== undefined){
				this.props.navigation.navigate('Home')
			}
		} catch (e) {
			alert(e);
		}
	}

	render() {
		if (!this.state.isReady) {
			return (
			  <AppLoading
				startAsync={this._loadAssetsAsync}
				onFinish={() => this.setState({ isReady: true })}
				onError={console.warn}
			  />
			);
		  }
		return (
			<View style={styles.container}>
				<ImageBackground source={require('../assets/initScreen.jpg')} style={styles.image}>
				</ImageBackground>
				<Image
					style={styles.tinyLogo}
					source={require('../assets/logo-wh.png')}
				/>
				<Text style={styles.text}>Let's Build</Text>
				<Text style={styles.subtext}>A Workout</Text>
				<TouchableOpacity style={styles.button} onPress={() => this.props.navigation.navigate('Login')}>
					<Text style={styles.buttonText}>Login</Text>
				</TouchableOpacity>
				<TouchableOpacity style={styles.buttonRight} onPress={() => this.props.navigation.navigate('Signup')}>
					<Text style={styles.buttonText}>Register</Text>
				</TouchableOpacity>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	tinyLogo: {
		position: 'absolute',
		top: vh(5),
        width: 100,
        height: 100
    },
	container: {
		backgroundColor: 'black',
		flex: 1,
		flexDirection: 'column',
	  },
	image: {
		flex: 1,
		resizeMode: 'cover',
		justifyContent: 'center',
		opacity:.7,
	  },
	button: {
		position: 'absolute',
		bottom: vh(5),
		left: vw(2.5),
		width: "40%",
		backgroundColor: "#0466C8",
		borderRadius: 20,
		height: vh(7),
		alignItems: "center",
		justifyContent: "center",
	},
	buttonRight: {
		position: 'absolute',
		bottom: vh(5),
		right: vw(2.5),
		width: "40%",
		backgroundColor: "#0466C8",
		borderRadius: 20,
		height: vh(7),
		alignItems: "center",
		justifyContent: "center",
	},
	text: {
		position: 'absolute',
		top: vh(35),
		marginLeft:20,
		fontWeight: '800',
		fontSize: vw(17),
		color: '#FFFFFF',
	  },
	subtext: {
		position: 'absolute',
		top: vh(45),
		marginLeft:20,
		fontWeight: '400',
		fontSize: vw(10),
		lineHeight: 45,
		color: '#E1DDDD',
	  },
	buttonText: {
		color: "white"
	},
});

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ updateEmail, updatePassword, fetchUserObj }, dispatch)
}

const mapStateToProps = state => {
	return {
		user: state.user
	}
}

export default connect(
    mapStateToProps,
	mapDispatchToProps
)(Initial)