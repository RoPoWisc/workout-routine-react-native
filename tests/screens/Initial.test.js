import React from 'react';
import renderer from 'react-test-renderer';
import mockStore from 'redux-mock-store';

import {Initial} from '../../screens/initial';


describe('Initial', () => {
    
    test('renders', () => {
        const render = renderer.create(<Initial />).toJSON();
        expect(render).toMatchSnapshot();
    })
    it('check function', () => {
        let loghand = renderer.create(<Initial />).getInstance();
        expect(loghand.componentDidMount()).toBeDefined();


        
    })
    
});

