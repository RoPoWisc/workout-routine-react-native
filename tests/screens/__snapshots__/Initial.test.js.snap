// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial renders 1`] = `null`;
